pkg/reconciler/channel/channel.go

```diff
@@ -0,0 +1,412 @@
+		//	// This is an update, so it means what exists did not match what is in the spec, even for create.
+		//	for i, ss := range channel.Status.Subscribers {
+		//		if ss.UID == s.UID {
+		//			channel.Status.Subscribers[i].ObservedGeneration = s.Generation
+		//		}
+		//	}
+		//	return nil // Signal a re-reconcile.
+		//} else if err != nil {
+		//	return err
+		//}
+		if !found {
+			// PullSubscription does not exist, that's ok, create it now.
+			ps, err := c.RunClientSet.PubsubV1alpha1().PullSubscriptions(channel.Namespace).Create(ps)
+			if err != nil {
+				c.Recorder.Eventf(channel, corev1.EventTypeWarning, "CreateSubscriberFailed", "Creating Subscriber %q failed", genName)
+				return err
+			}
+			c.Recorder.Eventf(channel, corev1.EventTypeNormal, "CreatedSubscriber", "Created Subscriber %q", ps.Name)
+		} else if !equality.Semantic.DeepEqual(ps.Spec, existingPs.Spec) {
+			ps, err := c.RunClientSet.PubsubV1alpha1().PullSubscriptions(channel.Namespace).Update(ps)
+			if err != nil {
+				c.Recorder.Eventf(channel, corev1.EventTypeWarning, "UpdateSubscriberFailed", "Updating Subscriber %q failed", genName)
+				return err
+			}
+			c.Recorder.Eventf(channel, corev1.EventTypeNormal, "UpdatedSubscriber", "Updated Subscriber %q", ps.Name)
+		}
+		for i, ss := range channel.Status.Subscribers {
+			if ss.UID == s.UID {
+				channel.Status.Subscribers[i].ObservedGeneration = s.Generation
+			}
+		}
+		return nil
+	}
+	for _, s := range subDeletes {
+		genName := resources.GenerateSubscriptionName(s.UID)
+		c.Logger.Infof("Channel %q will delete subscription %s", channel.Name, genName)
+		// TODO: we need to handle the case of a already deleted pull subscription. Perhaps move to ensure deleted method.
+		if err := c.RunClientSet.PubsubV1alpha1().PullSubscriptions(channel.Namespace).Delete(genName, &metav1.DeleteOptions{}); err != nil {
+			c.Logger.Errorf("unable to delete PullSubscription %s for Channel %q, %s", genName, channel.Name, err.Error())
+			c.Recorder.Eventf(channel, corev1.EventTypeWarning, "DeleteSubscriberFailed", "Deleting Subscriber %q failed", genName)
+			return err
+		}
+		c.Recorder.Eventf(channel, corev1.EventTypeNormal, "DeletedSubscriber", "Deleted Subscriber %q", genName)
+
+		for i, ss := range channel.Status.Subscribers {
+			if ss.UID == s.UID {
+				// Swap len-1 with i and then pop len-1 off the slice.
+				channel.Status.Subscribers[i] = channel.Status.Subscribers[len(channel.Status.Subscribers)-1]
+				channel.Status.Subscribers = channel.Status.Subscribers[:len(channel.Status.Subscribers)-1]
+			}
+		}
+		return nil // Signal a re-reconcile.
+	}
+
+	return nil
+}
+
+func (r *Reconciler) createTopic(ctx context.Context, channel *v1alpha1.Channel) (*pubsubv1alpha1.Topic, error) {
+	topic, err := r.getTopic(ctx, channel)
+	if err != nil && !apierrors.IsNotFound(err) {
+		logging.FromContext(ctx).Error("Unable to get an Topic", zap.Error(err))
+		return nil, err
+	}
+	if topic != nil {
```

Comment:
Should we check that we actually own this topic?

Reply:
... time passes... you actually do check the owner, but if it's not owned by us, you return a notfound, which will try to recreate, which will fail.

Reply:
Serving does a get, verifies the owner and fails the resource as 'XYZNotOwned'

Reply:
I throw not Owned now. Thanks again.